aip/general/0236.md

```diff
@@ -0,0 +1,321 @@
+has the following resource name pattern:
+
+`projects/{project}/locations/{location}/policies/{policy}`
+
+The experimental versions of the resource used for preview mode are represented
+as a nested collection under `Policy`. The resource type will follow the
+convention "<resource_name> + Experiment”. The following is the pattern for the
+experiment:
+
+`projects/{project}/locations/{location}/policies/{policy}/experiments/{experiment}`
+
+A proto used to represent an experiment **must** contain the following fields:
+
+  1. The required top-level fields for a resource, like `name` and `etag`
+  2. The policy that is being tested itself
+  3. The field, `preview_metadata` for preview
+
+
+```proto
+message PolicyExperiment {
+
+  // The name of the PolicyExperiment.
+  string name = 1;
+ 
+  // The policy experiment. This Policy will be used to simulate the effects of
+  // the change but will affect life traffic.
+  Policy policy = 2;
+
+  // The metadata associated with this policy experiment.
+  PreviewMetadata preview_metadata = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+- A new proto **must** be used for each experiment. There **must not** be a
+  common <ResourceName>Preview proto for all experiments.
+  - It **must not** use the same resource type as the live policy.
+- The experiment proto **must** contain a top-level field with the same type as
+   the live policy. 
+  - It **must** be named as the live resource type. For example, if the
+    experiment is for FirewallPolicy, then this field **must** be named
+    firewall_policy.
+- The name inside the embedded policy message **must** be the name of the live
+  policy. 
+- When the user is ready to promote an experiment, they **must** copy the 
+  `policy` message into the live policy and delete the experiment. This can be
+  done manually or via a "commit" method.
+- A product **may** support multiple simulated configurations concurrently in
+  preview for a single live policy. 
+  - This is represented using two or more resources in the nested collection.
+  - The number of experimental configuration for a given live policy **may** be
+    capped at a certain number.
+- Cascading deletes **must** occur: if the top-level resource is deleted, all
+  experiments **must** also be deleted.
+
+### Metadata
+A `preview_metadata` field is used to track all variables pertinent to a
+preview.
+ 
+```proto 
+message PreviewMetadata {
+  // State of the previewed resource. Possible values are ACTIVE and INACTIVE. 
+  // ACTIVE indicates that results are being logged downstream.
+  string state = 1;
+
+  // An identifying string common to all preview mode logs for the product.
+  // Searching logs for this string will isolate all logs pertaining to this
+  // preview.
+  string log_prefix = 2;
+
+  // The start time of this preview.
+  google.protobuf.Timestamp start_time = 3;
+
+  // The latest stop time of this preview.
```

Comment:
"Latest" as in "it **will** automatically be stopped at this point" or "latest" as in "when this **was** most recently stopped"?

Reply:
(I'm pretty sure it's "most recent", in which case I'd use that terminology, and also apply it to `start_time`.)